include/pybind11/pybind11.h

```diff
@@ -1199,6 +1206,8 @@ struct handle_type_name<cpp_function> {
 
 PYBIND11_NAMESPACE_END(detail)
 
+struct gil_not_used {};
```

Comment:
I like how much simpler it is. I'm not sure about the `mod`; if we put anything, it should have the full `module`, we don't use `mod` elsewhere for Python, and it isn't clearly module. But it can't be misplaced, and we don't "namespace" anything else like it.

I'm not sure it would actually be any simpler if we added a second option later (like some sort of multiphase init option). But maybe it's better to simplify now rather than worry too much about a future (which would not be more complex by this style either, at least).

`py::free_threaded()` seems like it would match the naming of CPython better, though most of the CPython internals related to the feature seem to refer to the GIL.

Reply:
> I like how much simpler it is.

It's simpler only here, but my suggested alternative is simpler overall, because it avoids the overloads and the `private` implementation of the `create_extension_module` function. For the user it doesn't make a difference.

My suggestion includes the `mod_` so that it comes out very similar to the existing name: `Py_MOD_GIL_NOT_USED` vs. py::mod_gil_not_used().

Another thought was that it's very easy to experiment, e.g. if

```
PYBIND11_MODULE(external_module, m, py::mod_gil_not_used())
```

has some issues that are difficult to resolve in the short term, it can be changed to

```
PYBIND11_MODULE(external_module, m, py::mod_gil_not_used(false))
```

and it'll be very obvious that free-threading support was attempted, and how to start trying again. It'll also be very easy to make the free-threading support platform-specific this way.

Reply:
> but my suggested alternative is simpler overall, because it avoids the overloads and the private implementation

That's what I meant by simpler.

Reply:
It's also called `PyUnstable_Module_SetGIL`, which does have the `module` in it. But I hadn't noticed how that mimics the slot value name.

Reply:
> That's what I meant by simpler.

Oh, got it. Thanks. (The way it appeared in the email message got me on the wrong track.)

> And `py::free_threaded(false)` avoids the double negative.

That works for me, too. — Usually I strive to not introduce new names if there is one already, but the double negative bothered me, too.

Reply:
Happy to have @colesbury have final say on what name he prefers.